packages/flutter/test/rendering/editable_test.dart

```diff
@@ -783,15 +732,15 @@ void main() {
     editable.selection = const TextSelection.collapsed(offset: 0);
     pumpFrame();
 
-    await simulateKeyDownEvent(LogicalKeyboardKey.arrowRight, platform: 'android');
-    await simulateKeyUpEvent(LogicalKeyboardKey.arrowRight, platform: 'android');
+    editable.moveSelectionRight(SelectionChangedCause.keyboard);
     expect(currentSelection.isCollapsed, true);
     expect(currentSelection.baseOffset, 1);
+    editable.selection = currentSelection;
```

Comment:
I think this line of code is redundant because it does not need to reset the selection.

Reply:
It will actually fail without this line.  The reason is that RenderEditable relies on EditableText to set the TextEditingController's selection when RenderEditableCalls onSelectionChanged.  So because I'm rendering a raw RenderEditable by itself here, I need to update the selection manually.

I'll add a comment about this because it is very confusing.  It took me awhile to figure this out while working on these tests.